pkg/server/sessions/internal/transactions/transactions.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transactions
+
+import (
+	"github.com/codenotary/immudb/embedded/sql"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/codenotary/immudb/pkg/database"
+	"sync"
+)
+
+type transaction struct {
```

Comment:
it feels rare transaction has not behaviour but just a holder

Reply:
transaction on server are needed to decoulple internal embedded transactions from the session logics.As instance more info are needed to control that only one read write transaction per session is executed